core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -11,25 +11,41 @@ import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicInteger
 import java.util.regex.Pattern
 
-import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, DeadLetterSuppression, NoSerializationVerificationNeeded, Props, Status, Terminated}
+import akka.actor.{ Actor, ActorLogging, ActorRef, Cancellable, Props, Status, Terminated }
 import akka.event.LoggingReceive
 import org.apache.kafka.clients.consumer._
-import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.{ Metric, MetricName, TopicPartition }
 import org.apache.kafka.common.errors.WakeupException
 
+import scala.collection.JavaConverters._
 import java.util.concurrent.locks.LockSupport
 
 import akka.Done
 
-import scala.util.control.{NoStackTrace, NonFatal}
+import scala.util.control.{ NoStackTrace, NonFatal }
 import scala.collection.JavaConverters._
 import scala.concurrent.duration._
 
 object KafkaConsumerActor {
-  case class StoppingException() extends RuntimeException("Kafka consumer is stopping")
+  final case class StoppingException() extends RuntimeException("Kafka consumer is stopping")
+
   def props[K, V](settings: ConsumerSettings[K, V]): Props =
     Props(new KafkaConsumerActor(settings)).withDispatcher(settings.dispatcher)
 
+  // ------------ public protocol ------------
+  /** DO NOT INHERIT */
+  sealed class RequestMetrics
+  case object RequestMetrics extends RequestMetrics {
+    /** Java DSL */
+    def getInstance: RequestMetrics = this
+  }
+  /** Contains a snapshot of metrics obtained from the underlying kafka consumer */
+  final case class ConsumerMetrics(metrics: Map[MetricName, Metric]) {
+    /** Java API */
+    def getMetrics: util.Map[MetricName, Metric] = metrics.asJava
+  }
+  // ------------ end of public protocol ------------
```

Comment:
This works when the KafkaConsumer actor is created and passed in, but what about the other cases where the actor is created by the library? Wouldn't it be better to expose the metrics via the `Control` materialized value?

Could be a Future and using ask internally.

Reply:
Yeah good point, will add that